rateLimiterLambda/main.go

```diff
@@ -0,0 +1,136 @@
+	FunctionNames []string `json:"functionNames"`
+}
+
+type InputData struct {
+	IntValue int `json:"intValue"`
+}
+
+var createFeatureFlagFunctionName string
+var getUserFeatureFlagFunctionName string
+var createUserFeatureFlagFunctionName string
+var getAllFeatureFlagsFunctionName string
+var getUserFeatureFlagsFunctionName string
+var updateFeatureFlagFunctionName string
+var getFeatureFlagFunctionName string
+var getUserFeatureFlagFunction string
+
+func init() {
+	env, found := os.LookupEnv(utils.ENV)
+	if !found {
+		log.Print("Env variable not set, making it by default PROD")
+		os.Setenv(utils.ENV, utils.PROD)
+	}
+	log.Printf("The env is %v", env)
+
+	createFeatureFlagFunctionName, found = os.LookupEnv("CreateFeatureFlagFunction")
+	if !found {
+		log.Println("Create feature flag function name not being set")
+	}
+
+	getUserFeatureFlagFunctionName, found = os.LookupEnv("GetUserFeatureFlagFunction")
+	if !found {
+		log.Println("Create feature flag function name not being set")
+	}
+
+	createUserFeatureFlagFunctionName, found = os.LookupEnv("CreateUserFeatureFlagFunction")
+	if !found {
+		log.Println("Create user feature flag function name not being set")
+	}
+
+	getUserFeatureFlagsFunctionName, found = os.LookupEnv("GetUserFeatureFlagsFunction")
+	if !found {
+		log.Println("get user feature flags function name not being set")
+	}
+
+	getAllFeatureFlagsFunctionName, found = os.LookupEnv("GetAllFeatureFlagFunction")
+	if !found {
+		log.Println("get all feature flag function name not being set")
+	}
+
+	updateFeatureFlagFunctionName, found = os.LookupEnv("UpdateFeatureFlagFunction")
+	if !found {
+		log.Println("Update feature flag function name not being set")
+	}
+
+	getFeatureFlagFunctionName, found = os.LookupEnv("GetFeatureFlagFunction")
+	if !found {
+		log.Println("get feature flag function name not being set")
+	}
+
+}
+
+func handler(ctx context.Context, event json.RawMessage) (events.APIGatewayProxyResponse, error) {
+	sess, err := session.NewSession()
+	if err != nil {
+		log.Println("Error in creation of AWS session, please contact on #feature-flag-service discord channel.")
+	}
+	lambdaClient := lambda.New(sess)
+
+	var inputData InputData
+	if err := json.Unmarshal(event, &inputData); err != nil {
+		return events.APIGatewayProxyResponse{
+			Body: "Unable to read input",
+			StatusCode: http.StatusBadRequest,
+		}, nil
+	}
+
+	var request = Request{
+		FunctionNames: []string{createFeatureFlagFunctionName,
+			createUserFeatureFlagFunctionName,
+			getFeatureFlagFunctionName,
+			getUserFeatureFlagFunctionName,
+			getAllFeatureFlagsFunctionName,
+			updateFeatureFlagFunctionName,
+			getUserFeatureFlagsFunctionName,
+		},
+	}
+
+	log.Printf("The request data is %d", inputData.IntValue)
+
+	for _, functionName := range request.FunctionNames {
+		input := &lambda.PutFunctionConcurrencyInput{
+			FunctionName:                 &functionName,
+			ReservedConcurrentExecutions: aws.Int64(int64(inputData.IntValue)),
+		}
```

Comment:
please follow up with the discussion done over the call over spanning to multiple lambdas

Reply:
Had discussion with Ankush, this is the view point shared

> We can skip auto-scaling right now, but we should monitor how many concurrent calls are we getting, so we are aware of our changing max limits
We should have an alarm if we go beyond

cc : @RitikJaiswal75